src/service/blockchain-event-listener-service.js

```diff
@@ -118,14 +118,13 @@ class BlockchainEventListenerService {
     async handleBlockchainEvents(events) {
         if (events?.length) {
             this.logger.trace(`${events.length} blockchain events caught.`);
-            const insertedEvents = await this.repositoryModuleManager.insertBlockchainEvents(
-                events,
-            );
-            insertedEvents.sort((event1, event2) => event1.block - event2.block);
+            await this.repositoryModuleManager.insertBlockchainEvents(events);
+            const unprocessedEvents =
+                await this.repositoryModuleManager.getAllUnprocessedBlockchainEvents();
```

Comment:
I would rather change the code not to store those events that we are not processing. We need to fetch all events because of that issue if node restarts during event handling, we wan't to fetch everything that is not processed from the table. We wan't to make sure that all of them are processed